docs/development.rst

```diff
@@ -244,37 +261,74 @@ This plugin produces a coverage report, example::
 
 
+Profiling
++++++++++
+
 Profiling can be done by e.g.::
 
-   $ python -m cProfile  -s time ./mslib/mswms/demodata.py > profile.txt
+   $ python -m cProfile  -s time ./mslib/mswms/demodata.py --seed > profile.txt
 
 example::
 
-    /!\ existing server config: "mss_wms_settings.py" for demodata not overwritten!
+   /!\ existing server config: "mss_wms_settings.py" for demodata not overwritten!
 
 
-    To use this setup you need the mss_wms_settings.py in your python path e.g.
-    export PYTHONPATH=$HOME/mss
-             398119 function calls (389340 primitive calls) in 0.834 seconds
+   /!\ existing server auth config: "mss_wms_auth.py" for demodata not overwritten!
 
-       Ordered by: internal time
 
-       ncalls  tottime  percall  cumtime  percall filename:lineno(function)
-           19    0.124    0.007    0.496    0.026 demodata.py:912(generate_file)
-           19    0.099    0.005    0.099    0.005 {method 'close' of 'netCDF4._netCDF4.Dataset' objects}
+   To use this setup you need the mss_wms_settings.py in your python path e.g.
+   export PYTHONPATH=~/mss
+         557395 function calls (543762 primitive calls) in 0.980 seconds
 
+   Ordered by: internal time
 
+   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
+       23    0.177    0.008    0.607    0.026 demodata.py:1089(generate_file)
+      631    0.113    0.000    0.230    0.000 demodata.py:769(_generate_3d_data)
+      179    0.077    0.000    0.081    0.000 {method 'createVariable' of 'netCDF4._netCDF4.Dataset' objects}
 
-Setup mss_settings.json
-----------------------------
 
-On default all tests use default configuration defined in mslib.msui.MissionSupportSystemDefaultConfig.
-If you want to overwrite this setup and try out a special configuration add an mss_settings.json
-file to the testings base dir in your tmp directory.
+
+Pushing your changes
+--------------------
+
+1. Now you have made the changes, tested them and built them. So now it's time to push them.
+2. Goto your terminal and type git status and hit enter, this will show your changes from the files
+3. Then type in git add and hit enter, this will add all the files to staging area
+4. Commit the changes by ``git commit -m "<message-describing-your-change>"`` and hit enter.
+5. Now push your branch to your fork by ``git push origin <your-branch-name>`` and hit enter.
 
 
+Creating a pull request
+-----------------------
+
+By this time you can see a message on your github fork as your fork is ahead of Open-MSS:develop by <number> of commits and also you can see a button called Compare and pull request.
+
+Click on Compare and pull request button.
+
+You will see a template.
+
+Fill out the template completely by describing your change, cause of change, issue getting fixed etc.
+
+After filling the template completely click on Pull request
+
+
+Guides
+~~~~~~
+
+Coding Style
+------------
+
+We generally follow flake8, with 120 columns instead of 79.
```

Comment:
flake8 is not a style guide...? PEP8 (with link https://www.python.org/dev/peps/pep-0008/) ?
Perhaps we follow PEP8 and check for adherance to this standard using the flake8 tool.